check-vault-status.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+#
+# check-vault-status
+# 
+# DESCRIPTION:
+#   This plugins checks if vault is up & reachable. It then checks
+#   if the vault is unsealed
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: vault
+#
+require 'sensu-plugin/check/cli'
+require 'vault'
+
+class VaultStatus < Sensu::Plugin::Check::CLI
+
+  option :host,
+    description: 'vault host',
+    short: '-h HOST',
+    long:  '--host HOST',
+    default: '127.0.0.1'
+
+
+  option :port,
+    description: 'vault port',
+    short: '-p PORT',
+    long: '--port PORT',
+    default: '8200'
+
+  option :scheme,
```

Comment:
I tend to prefer `protocol` over `scheme` as in the context of metric checks it can be easily mistaken for the scheme/format of the metric path.